android/src/app/organicmaps/widget/menu/NavMenu.java

```diff
@@ -239,4 +246,22 @@ public void update(@NonNull RoutingInfo info)
 
     void onSettingsClicked();
   }
+
+  private void updateRouteAltitudeChart(int state)
+  {
+    if (state == BottomSheetBehavior.STATE_COLLAPSED) {
+      mRoutingChart.setVisibility(View.GONE);
+    } else {
+      mRoutingChart.setVisibility(View.VISIBLE);
+      if (mRoutingChart.getDrawable() == null) {
+        int chartWidth = ((View) mRoutingChart.getParent()).getWidth();
+        int chartHeight = UiUtils.dimen(mActivity, R.dimen.altitude_chart_image_height);
+        Framework.RouteAltitudeLimits limits = new Framework.RouteAltitudeLimits();
+        Bitmap bm = Framework.generateRouteAltitudeChart(chartWidth, chartHeight, limits);
+        if (bm != null) {
```

Comment:
1. When we can have null here? Should we crash to debug and fix it earlier?
2. If error case is a valid one, should the view be in the GONE state?

Reply:
1.- The chart should now update for new routes searches. I didn't realised views where not recreated from scratch every time but saved.
2.- In case of NULL bm the chart is now GONE